clippy_lints/src/unused_io_amount.rs

```diff
@@ -45,126 +46,241 @@ declare_clippy_lint! {
 
 declare_lint_pass!(UnusedIoAmount => [UNUSED_IO_AMOUNT]);
 
+#[derive(Copy, Clone)]
+enum IoOp {
+    AsyncWrite(bool),
+    AsyncRead(bool),
+    SyncRead(bool),
+    SyncWrite(bool),
+}
+
+/// We need to check that read/write call sizes are handled.
+/// We need to be careful with this to avoid false positives.
+/// If we do the check directly on Expr, we will miss cases like:
+/// `
```

Comment:
```suggestion
///
```

Typo? :)

Reply:
Ah wait, I'm not sure it's a typo. If you want a code block, why not use \`\`\`?

Reply:
I think the comments need to be cleaned up a bit 😬